src/app_service/service/profile/service.nim

```diff
@@ -1,80 +1,147 @@
+        error "error: ", procName="storeIdentityImage", errDesription = "response is not an array"
+        return
+      if(response.result.len == 0):
+        error "error: array of stored images is empty"
+        return
 
-    for img in response.result:
-      let imageDto = toImage(img)
-      result.add(imageDto)
-      if(imageDto.imgType == "large"):
-        singletonInstance.userProfile.setLargeImage(imageDto.uri)
-      elif(imageDto.imgType == "thumbnail"):
-        singletonInstance.userProfile.setThumbnailImage(imageDto.uri)
-
-  except Exception as e:
-    error "error: ", procName="storeIdentityImage", errName = e.name, errDesription = e.msg
-
-proc deleteIdentityImage*(self: Service, address: string) =
-  try:
-    let response = status_accounts.deleteIdentityImage(address)
-    if(not response.error.isNil):
-      error "could not delete identity images"
-      return
-    singletonInstance.userProfile.setLargeImage("")
-    singletonInstance.userProfile.setThumbnailImage("")
+      for img in response.result:
+        let imageDto = toImage(img)
+        result.add(imageDto)
+        if(imageDto.imgType == "large"):
+          singletonInstance.userProfile.setLargeImage(imageDto.uri)
+        elif(imageDto.imgType == "thumbnail"):
+          singletonInstance.userProfile.setThumbnailImage(imageDto.uri)
 
-  except Exception as e:
-    error "error: ", procName="deleteIdentityImage", errName = e.name, errDesription = e.msg
+    except Exception as e:
+      error "error: ", procName="storeIdentityImage", errName = e.name, errDesription = e.msg
 
-proc setDisplayName*(self: Service, displayName: string) =
-  try:
-    let response =  status_accounts.setDisplayName(displayName)
-    if(not response.error.isNil):
-      error "could not set display name"
-      return
-    if(not self.settingsService.saveDisplayName(displayName)):
-      error "could save display name to the settings"
+  proc deleteIdentityImage*(self: Service, address: string) =
+    try:
+      let response = status_accounts.deleteIdentityImage(address)
+      if(not response.error.isNil):
+        error "could not delete identity images"
+        return
+      singletonInstance.userProfile.setLargeImage("")
+      singletonInstance.userProfile.setThumbnailImage("")
+
+    except Exception as e:
+      error "error: ", procName="deleteIdentityImage", errName = e.name, errDesription = e.msg
+
+  proc setDisplayName*(self: Service, displayName: string) =
+    try:
+      let response =  status_accounts.setDisplayName(displayName)
+      if(not response.error.isNil):
+        error "could not set display name"
+        return
+      if(not self.settingsService.saveDisplayName(displayName)):
+        error "could save display name to the settings"
+        return
+    except Exception as e:
+      error "error: ", procName="setDisplayName", errName = e.name, errDesription = e.msg
+
+  proc requestProfileShowcasePreferences*(self: Service) =
+    try:
```

Comment:
The try is not really necessary since we only start the threadpool here.

Reply:
fixed